lib/reporters/default.js

```diff
@@ -0,0 +1,1125 @@
+'use strict';
+const os = require('os');
+const path = require('path');
+const stream = require('stream');
+
+const cliCursor = require('cli-cursor');
+const figures = require('figures');
+const indentString = require('indent-string');
+const ora = require('ora');
+const plur = require('plur');
+const prettyMs = require('pretty-ms');
+const trimOffNewlines = require('trim-off-newlines');
+
+const chalk = require('../chalk').get();
+const codeExcerpt = require('../code-excerpt');
+const beautifyStack = require('./beautify-stack');
+const colors = require('./colors');
+const formatSerializedError = require('./format-serialized-error');
+const improperUsageMessages = require('./improper-usage-messages');
+const prefixTitle = require('./prefix-title');
+const whileCorked = require('./while-corked');
+
+const nodeInternals = require('stack-utils').nodeInternals();
+
+class LineWriter extends stream.Writable {
+	constructor(dest, spinner) {
+		super();
+
+		this.dest = dest;
+		this.columns = dest.columns || 80;
+		this.lastLineIsEmpty = false;
+		this.lastSpinnerText = '';
+		this.spinner = spinner;
+	}
+
+	_write(chunk, encoding, callback) {
+		if (this.spinner) {
+			this.spinner.clear();
+			this._writeWithSpinner(chunk.toString('utf8'));
+		} else {
+			this.dest.write(chunk);
+		}
+
+		callback();
+	}
+
+	_writev(pieces, callback) {
+		if (!this.spinner) {
+			return;
```

Comment:
It might be better to have two writers, one with, and one without a spinner. There are a lot of unstated assumptions in the `if` / `else` conditions that are to do with how the reporter is instantiated.